### PR TITLE
Merge tag 'swift-DEVELOPMENT-SNAPSHOT-2019-07-28-a' into tensorflow

### PR DESCRIPTION
**Pull request by dan-zheng**

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
clean
@swift-ci Please test tensorflow

File: utils/build-presets.ini [hunk 1]
```diff
@@ -2103,26 +2103,33 @@ swiftpm
 swiftsyntax
 skstresstester
 playgroundsupport
+libcxx
```
Comment (dan-zheng) [inline utils/build-presets.ini]: I wonder if adding these flags increases the toolchain size. Also `install-libcxx` and additions to `swift-install-components` and `llvm-install-components` below.

**Comment by dan-zheng** [comment]
Confirmed that `build-toolchain-tensorflow` succeeds on Ubuntu 18.04 and macOS.
Ready to merge.

---

With `build-toolchain-tensorflow -t`: all Swift tests pass, but lldb tests are unresolved, which I believe is known and not a regression. cc @bgogul @marcrasi

```
# lldb test results.
  Expected Passes    : 34
  Unsupported Tests  : 24
  Unresolved Tests   : 227
```

Example unresolved test:
```
********************
UNRESOLVED: lldb-Suite :: lang/swift/variables/string/TestSwiftStringVariables.py (255 of 285)
******************** TEST 'lldb-Suite :: lang/swift/variables/string/TestSwiftStringVariables.py' FAILED ********************
Unable to find 'RESULT: PASSED' in dotest output:

lldb version 7.0.0 (https://github.com/apple/swift-lldb.git revision e95fced8b55d5c2b3a99fa12f61f3d8f1cdb9f5a)
Swift version 5.1-dev (LLVM 200186e28b, Swift d3a26fba10)
LLDB library dir: /home/danielzheng/swift-dev/build/buildbot_linux/lldb-linux-x86_64/bin
LLDB import library dir: /home/danielzheng/swift-dev/build/buildbot_linux/lldb-linux-x86_64/bin
Traceback (most recent call last):
  File "/home/danielzheng/swift-dev/lldb/test/dotest.py", line 7, in <module>
    lldbsuite.test.run_suite()
  File "/home/danielzheng/swift-dev/lldb/packages/Python/lldbsuite/test/dotest.py", line 1272, in run_suite
    import lldb
  File "/home/danielzheng/swift-dev/build/buildbot_linux/lldb-linux-x86_64/lib/python3.7/site-packages/lldb/__init__.py", line 38, in <module>
    from . import _lldb
ImportError: dynamic module does not define init function (init_lldb)
```

This may be due to an lldb/Python configuration problem.

**Review by pschuh — APPROVED** [review]

**Pull request merged**